### PR TITLE
feat: foxy APR RTK query

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This removes the `useFoxyApr` logic to an RTK query, consumed directly from components

- [x] implement `getFoxyApr` hook
- [x] consume hook instead of `useFoxyApr`
- [x] remove `useFoxyApr` hook and tests
- [x] update CSP comments
- [ ] rename `foxyBalancesApi` to `foxyApi` to make things more semantic now that this API is not responsible for balances only (TBD as the last PR of the stack to ease mid-stack updates in case changes are requested)

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

- All places with FOXy APR could have regressions - tested across the FOX page and DeFi and didn't find any.

## Testing

Test that FOXy apr is still loaded and displayed:

- FOX Page
- DeFi Earn
- DeFi overview
- FOXy DeFi modal

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2512** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2512" target="_blank">Graphite</a> 
    * **PR #2523** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2523" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2523?utm_source=stack-comment).

File: src/state/apis/foxy/foxyBalancesApi.ts [hunk 1]
```diff
@@ -220,7 +239,40 @@ export const foxyBalancesApi = createApi({
         }
       },
     }),
+    getFoxyApr: build.query<GetFoxyAprOutput, void>({
+      queryFn: async () => {
+        const response = await axios.get<{ chains: TokemakChainData[] }>(TOKEMAK_STATS_URL)
```
Comment (gomesalexandre) [inline src/state/apis/foxy/foxyBalancesApi.ts:244]: If we want to be extra accurate, we should handle fetching errors and return the error code as a `{status, data}` shape

File: src/state/apis/foxy/foxyBalancesApi.ts [hunk 2]
```diff
@@ -55,6 +65,16 @@ const moduleLogger = logger.child({
 
 type MaybeMarketCapData = Record<AssetId, MarketData | undefined>
 
+type TokemakPool = {
+  address: string
+  liquidityProviderApr: string
+}
+
+type TokemakChainData = {
+  chainId: string
```
Comment (0xdef1cafe) [inline src/state/apis/foxy/foxyBalancesApi.ts:74]:
```suggestion
  chainId: ChainId
```

Comment (0xdef1cafe) [inline src/state/apis/foxy/foxyBalancesApi.ts:74]:
```suggestion
  chainId: string // EVM chainId, not CAIP ChainId
```

**Review by 0xdef1cafe — DISMISSED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**